sdk/react-native/src/ReactNativeCacheStore.ts

```diff
@@ -0,0 +1,115 @@
+import AsyncStorage from '@react-native-async-storage/async-storage'
+import { DVCCacheStore, StoreKey } from '@devcycle/devcycle-js-sdk'
+import { BucketedUserConfig, DVCLogger } from '@devcycle/types'
+import { DVCPopulatedUser } from 'sdk/js/src/User'
+
+export class ReactNativeStore implements DVCCacheStore {
+    store: typeof AsyncStorage
+    logger?: DVCLogger
+
+    constructor(logger?: DVCLogger) {
+        this.store = AsyncStorage
+        this.logger = logger
+    }
+
+    save(storeKey: string, data: unknown): void {
+        this.store.setItem(storeKey, JSON.stringify(data))
+          .catch(err => {
+            this.logger?.debug(`Error saving ${storeKey} to local storage: ${JSON.stringify(err)}`)
+          })
+    }
+
+    load<T>(storeKey: string): Promise<T | null | undefined> {
+      return this.store.getItem(storeKey)
+        .then(item => {
+          return (item ? JSON.parse(item) : null)
+        })
+        .catch(err => {
+          this.logger?.debug(`Error loading ${storeKey} from local storage: ${JSON.stringify(err)}`)
+          return null
+        })
+    }
+
+    private getConfigKey(user: DVCPopulatedUser) {
+        return user.isAnonymous ? StoreKey.AnonymousConfig : StoreKey.IdentifiedConfig
+    }
+
+    private getConfigUserIdKey(user: DVCPopulatedUser) {
+        return `${this.getConfigKey(user)}.user_id`
+    }
+
+    private getConfigFetchDateKey(user: DVCPopulatedUser) {
+        return `${this.getConfigKey(user)}.fetch_date`
+    }
+
+    private async loadConfigUserId(user: DVCPopulatedUser): Promise<string | null | undefined> {
+        const userIdKey = this.getConfigUserIdKey(user)
+        return this.load<string>(userIdKey)
+    }
+
+    private async loadConfigFetchDate(user: DVCPopulatedUser): Promise<number> {
+        const fetchDateKey = this.getConfigFetchDateKey(user)
+        return parseInt(await this.load<string>(fetchDateKey) || '0', 10)
+    }
+
+    remove(storeKey: string): void {
+        this.store.removeItem(storeKey)
+    }
+
+    saveConfig(data: BucketedUserConfig, user: DVCPopulatedUser, dateFetched: number): void {
+        const configKey = this.getConfigKey(user)
+        const fetchDateKey = this.getConfigFetchDateKey(user)
+        const userIdKey = this.getConfigUserIdKey(user)
+        this.save(configKey, data)
+        this.save(fetchDateKey, dateFetched)
+        this.save(userIdKey, user.user_id)
+        this.logger?.info('Successfully saved config to local storage')
+    }
+
+    async loadConfig(user: DVCPopulatedUser, configCacheTTL= 604800000): Promise<BucketedUserConfig | null> {
```

Comment:
is any of this stuff platform specific or could it go in a shared base class?

Reply:
only the `save` / `load` / `remove` are platform specific... 🤔 the rest are the same

Reply:
so can they be removed then?

Reply:
yeah, good point.. should be able to remove them :)

Reply:
the underlying save/load methods should be in their own Storage class that is separate from all this higher-level logic that uses that Storage class..otherwise anybody providing their own implementation of storage would have to re-implement or extend all these things

Reply:
yeah, I'm updating the `DVCCacheStore` interface to just have `save`, `load` and `remove`... and then the rest is implemented by the JS SDK CacheStore... because these 3 methods will be the only ones that would be Cache Store specific implementations

Reply:
yeah but the JS SDK `CacheStore` should basically have some pluggable `Storage` class that it calls to do `save / load`, which is the thing that this library would be replacing